### ISSUE TITLE
ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs

### ISSUE THREAD

**Issue by MatthewPringle** [dpa99c/cordova-plugin-firebasex#266]
Having included the plugin and submitted to Apple's App Store for Testflight I am getting the following email from Apple.

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app. Your delivery was successful, but you may wish to correct the following issues in your next delivery:
> 
> ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs . See https://developer.apple.com/documentation/uikit/uiwebview for more information.
> 
> After you’ve corrected the issues, you can upload a new binary to App Store Connect.
> 
> Best regards,
> 
> The App Store Team

**Comment by dpa99c** [comment]
This has nothing to do with this plugin and relates to the use of UIWebView by Cordova - see https://github.com/apache/cordova-ios/issues/661

**Comment by MatthewPringle** [comment]
I believe it is related to the version of Firebase being installed by the plugin rather than Cordova.

The app submits fine before I add the plugin and afterwards generates the error. I am on the latest Cordova.

See https://github.com/firebase/firebase-ios-sdk/pull/3723

**Comment by Sagarkhenat** [comment]
@MatthewPringle I think it is due to the firebase SDK's used by the plugin. I am also building an ionic app using ionic 3 and was using Cordova-plugin-firebase and got the warning mail. When I removed that plugin, the warning mail was not generated. I was going to use this plugin instead of that but I hope @dpa99c will share necessary information.

**Comment by MatthewPringle** [comment]
@Sagarkhenat yeah Firebase is a mess all round and im shocked about the lack of support Google provide in keeping it up to date.

The javascript version is even worse and refuses to run in Cordova unless you install a bunch of abandoned plugins which really no longer work and cause more emails from Apple about outdated code.

It is almost like they don't want you using Firebase!

**Comment by levieu** [comment]
> Having included the plugin and submitted to Apple's App Store for Testflight I am getting the following email from Apple.
> 
> > Dear Developer,
> > We identified one or more issues with a recent delivery for your app. Your delivery was successful, but you may wish to correct the following issues in your next delivery:
> > ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs . See https://developer.apple.com/documentation/uikit/uiwebview for more information.
> > After you’ve corrected the issues, you can upload a new binary to App Store Connect.
> > Best regards,
> > The App Store Team

I fixed it by following these steps (comment by Matt Netkow, Ionic Team in https://cordova.apache.org/announcements/2019/11/25/cordova-ios-release-5.1.0.html):
- Update cordova-ios at version 5.1.1
- Update cordova-plugin-ionic-webview at version 4.1.3
- Add <preference name="WKWebViewOnly" value="true"/> to your config.xml file.
- Run "cordova prepare ios" to apply the changes.

If you use the cordova-plugin-inappbrowser plugin update it to the version 3.2.0 (https://github.com/apache/cordova-plugin-inappbrowser/pull/584)

**Comment by dpa99c** [comment]
TL;DR: 
- Follow steps above 
- Update to latest version of this plugin
